planning/behavior_velocity_planner/include/scene_module/dynamic_obstacle_stop/scene.hpp

```diff
@@ -0,0 +1,170 @@
+
+  boost::optional<DynamicObstacle> detectCollision(
+    const std::vector<DynamicObstacle> & dynamic_obstacles,
+    const PathWithLaneId & path_points) const;
+
+  float calcCollisionPositionOfVehicleSide(
+    const geometry_msgs::msg::Point & point, const geometry_msgs::msg::Pose & base_pose) const;
+
+  std::vector<geometry_msgs::msg::Point> createVehiclePolygon(
+    const geometry_msgs::msg::Pose & base_pose) const;
+
+  std::vector<DynamicObstacle> checkCollisionWithObstacles(
+    const std::vector<DynamicObstacle> & dynamic_obstacles,
+    std::vector<geometry_msgs::msg::Point> poly, const float travel_time) const;
+
+  boost::optional<DynamicObstacle> findNearestCollisionObstacle(
+    const PathWithLaneId & path, const geometry_msgs::msg::Pose & base_pose,
+    std::vector<DynamicObstacle> & dynamic_obstacles) const;
+
+  boost::optional<geometry_msgs::msg::Pose> calcPredictedObstaclePose(
+    const std::vector<PredictedPath> & predicted_paths, const float travel_time,
+    const float velocity_mps) const;
+
+  bool checkCollisionWithShape(
+    const tier4_autoware_utils::Polygon2d & vehicle_polygon, const PoseWithRange pose_with_range,
+    const Shape & shape, std::vector<geometry_msgs::msg::Point> & collision_points) const;
+
+  bool checkCollisionWithCylinder(
+    const tier4_autoware_utils::Polygon2d & vehicle_polygon, const PoseWithRange pose_with_range,
+    const float radius, std::vector<geometry_msgs::msg::Point> & collision_points) const;
+
+  bool checkCollisionWithBoundingBox(
+    const tier4_autoware_utils::Polygon2d & vehicle_polygon, const PoseWithRange pose_with_range,
+    const geometry_msgs::msg::Vector3 & dimension,
+    std::vector<geometry_msgs::msg::Point> & collision_points) const;
+
+  bool checkCollisionWithPolygon() const;
+
+  std::vector<geometry_msgs::msg::Point> createBoundingBoxForRangedPoints(
+    const PoseWithRange & pose_with_range, const float x_offset, const float y_offset) const;
+
+  boost::optional<geometry_msgs::msg::Pose> calcStopPoint(
+    const boost::optional<DynamicObstacle> & dynamic_obstacle, const PathWithLaneId & path,
+    const geometry_msgs::msg::Pose & current_pose, const float current_vel,
+    const float current_acc) const;
+
+  void insertStopPoint(
+    const boost::optional<geometry_msgs::msg::Pose> stop_point,
+    autoware_auto_planning_msgs::msg::PathWithLaneId & path);
+
+  void insertVelocityWithApproaching(
+    const boost::optional<DynamicObstacle> & dynamic_obstacle,
+    const geometry_msgs::msg::Pose & current_pose, const float current_vel, const float current_acc,
+    const PathWithLaneId & resampled_path, PathWithLaneId & output_path);
+
+  void insertApproachingVelocity(
+    const DynamicObstacle & dynamic_obstacle, const geometry_msgs::msg::Pose & current_pose,
+    const float approaching_vel, const float approach_margin, const PathWithLaneId & resampled_path,
+    PathWithLaneId & output_path);
```

Comment:
The names of these two functions are confusing.
Can't "insertVelocityWithApproaching" be named something else, like "insertVelocity"? This function also execute "insertStopPoint".

Reply:
I renamed function name to "insertVelocity", and create the function "insertStoppingVelocity" to be easier to understand.
https://github.com/autowarefoundation/autoware.universe/pull/752/commits/cad055904b78ad379be79f0f0d6805d0701df000